rdkit/Chem/Draw/IPythonConsoleIntegration.py

```diff
@@ -0,0 +1,558 @@
+        wgData[uid]['prop_wg'] = Dropdown(description='',options=['select'], value='select')
+        wgBox.append(Box([Label(value='prop'),wgData[uid]['prop_view']], layout=itemLayout))
+        wgBox.append(Box([Label(value='prop'),wgData[uid]['prop_wg']], layout=itemLayout))
+        widgetsFor3DView.append('prop_wg')
+        
+    # useDrawAs
+    if useDrawAs:
+        if moldict is not None:
+            wgData[uid]['drawAs_wg'] = Dropdown(description='', options=DRAWING_LIGAND_3D, value=drawAs)
+            wgBox.append(Box([Label(value='drawAs'),wgData[uid]['drawAs_wg']], layout=itemLayout))
+            widgetsFor3DView.append('drawAs_wg')
+        if protein is not None:
+            wgData[uid]['pStyle_wg'] = Dropdown(description='', options=DRAWING_PROTEIN_3D, value=pStyle)
+            wgBox.append(Box([Label(value='pStyle'),wgData[uid]['pStyle_wg']], layout=itemLayout))
+            wgData[uid]['pStyle_helicesAsTubes_wg'] = Checkbox(description='helicesAsTubes', value=False)
+            wgBox.append(Box([Label(value=''),wgData[uid]['pStyle_helicesAsTubes_wg']], layout=itemLayout))
+            widgetsFor3DView.extend(['pStyle_wg', 'pStyle_helicesAsTubes_wg'])
+            
+    # colors
+    if colorPanel is True:
+        wgData[uid]['colorScheme'] = Dropdown(description='', options=LIGAND_COLOR_SCHEME_3D, value='default')
+        wgBox.append(Box([Label(value='ligand color'),wgData[uid]['colorScheme']], layout=itemLayout))
+        widgetsFor3DView.append('colorScheme')
+        
+    # labels
+    if labelPanel is True:
+        wgData[uid]['confLabel'] = Checkbox(description='confLabel', value=False)
+        wgData[uid]['atomLabel'] = Checkbox(description='atomLabel', value=False)
+        cbLabel=Box([Label(value=''), HBox([wgData[uid]['confLabel'], 
+                                            wgData[uid]['atomLabel']
+                                           ])
+                    ], layout=itemLayout)
+        wgBox.append(cbLabel)
+        widgetsFor3DView.extend(['confLabel', 'atomLabel'])
+        
+    # background
+    wgData[uid]['background'] = Dropdown(description='', options= BGCOLORS_3D, value=BGCOLORS_3D[0])
+    wgBox.append(Box([Label(value='background'),wgData[uid]['background']], layout=itemLayout))
+    widgetsFor3DView.append('background')
+    
+    
+    # buttons
+    wgData[uid]['start'] = Button(description="Start!", button_style='success')
+    wgData[uid]['zoomTo'] = Button(description="zoomTo", button_style='success')
+    
+    buttons=Box([Label(value=''),HBox([wgData[uid]['zoomTo'], wgData[uid]['start']])],layout=itemLayout)
+    wgBox.append(buttons)
+    
+    
+    # Observe and update dict with model_id
+    wgData[uid]['start'].on_click(handle_start_button)
+    wgData[uid]['zoomTo'].on_click(handle_zoomTo_button)
+    wgMIds[wgData[uid]['start'].model_id] = uid
+    wgMIds[wgData[uid]['zoomTo'].model_id] = uid
+    
+    wgDataUid = wgData[uid]
```

Comment:
The new codes does not have the line `wgDataUid = wgData[uid]`. I hope the follwing is readable too. Kindly let me know if you don't prefer it. I am willing to follow you instruction once again. Thank you very much for commenting on variable names!
```
for i in widgetsFor3DView:
            selectedWidget = wgData[uid][i]
            selectedWidget.observe(self.handle_change, names='value')
            
```